dace/data.py

```diff
@@ -33,7 +33,7 @@ def create_datadescriptor(obj, no_custom_desc=False):
         return obj.__descriptor__()
     elif not no_custom_desc and hasattr(obj, 'descriptor'):
         return obj.descriptor
-    elif dtypes.is_array(obj):
+    elif dtypes.is_array(obj) and (hasattr(obj, '__array_interface__') or hasattr(obj, '__cuda_array_interface__')):
```

Comment:
What regression does this cause?

Reply:
For torch tensors ``is_array`` is true, but ``__array_interface__`` doesn't exist. Thus this branch gets chosen but fails later when `__array_interface__`` is accessed.